dotnetv3/dynamodb/FromSQL/README.md

```diff
@@ -0,0 +1,415 @@
+
+The default table name is defined as **Table**
+in **app.config**.
+
+### Adding an item to the table
+
+Use the **AddItem** project to add an item to a table.
+
+The default table name is defined as **Table**
+and the default Region is defined as **Region**
+in **app.config**.
+
+It requires the following options:
+
+- ```-k``` *KEYS*, where *KEYS* is a a comma-separated list of keys
+- ```-v``` *VALUES*, where *VALUES* is a a comma-separated list of values
+
+There must be the same number of keys as values.
+
+- If the item is a customer, the schema should match that in **customers.csv**,
+  with one additional key, ID, which defines the partition ID for the item.
+- If the item is an order, the schema should match that in **orders.csv**,
+  with one additional key, ID, which defines the partition ID for the item*.
+- If the item is a product, the schema should match that in **products.csv**,
+  with one additional key, ID, which defines the partition ID for the item*.
+
+It's up to you to determine the appropriate partition key value (ID).
+If you provide the same value as an existing table item,
+the values of that item are overwritten.
+
+### Uploading items to a table
+
+Use the **AddItems** project to incorporate data from a comma-separated value
+(.csv) files to populate a table.
+
+The default table name is defined as **Table**,
+the default Region is defined as **Region**,
+and the default table names are defined as
+**Customers**, **Orders**, and **Products**
+in **app.config**.
+
+### Managing indexes
+
+Global secondary indices give you the ability to treat a set of
+Amazon DynamoDB table keys as if they were a separate table.
+
+#### Creating an index
+
+Use the **CreateIndex** project to create an index.
+
+The default table name is defined as **Table**,
+and the default Region is defined as **Region**
+in **app.config**.
+
+It requires the following command-line options:
+
+- ```-i``` *INDEX-NAME*, where *INDEX-NAME* is the name of the index
+- ```-m``` *MAIN-KEY*, where *MAIN-KEY* is the partition key of the index
+- ```-k``` *MAIN-KEY-TYPE*, where *MAIN-KEY-TYPE* is one of string or number
+- ```-s``` *SECONDARY-KEY*, where *SECONDARY-KEY* is the sort key of the index
+- ```-t``` *SECONDARY-KEY-TYPE*, where *SECONDARY-KEY-TYPE* is one of string or number
+
+To create a global secondary index (GSI) for the three queries we made in
+*A simple example*,
+execute the following commands:
+
+```
+CreateIndex.exe -i OrderDate       -m Area -k string -s Order_Date       -t number
+CreateIndex.exe -i ProductOrdered  -m Area -k string -s Order_Product    -t number
+CreateIndex.exe -i LowProduct      -m Area -k string -s Product_Quantity -t number
+```
+
+Note that you cannot execute these commands one after another.
+You must wait until one GSI is created before you can attempt to create another GSI.
+We recommend you use the Amazon DynamoDB console to monitor the progress
+of creating a GSI to avoid errors.
```

Comment:
**Note** 
You must wait between running each command for each GSI to be created. You can't create another GSI until the previous GSI is created. To avoid errors, we recommend using the DynamoDB console to monitor GSI creation.

Reply:
Fixed.